app/services/user_creator.rb

```diff
@@ -1,18 +1,19 @@
 # frozen_string_literal: true
 
 class UserCreator
-  def initialize(user_params:, provider:)
+  def initialize(user_params:, provider:, role:)
     @user_params = user_params
     @provider = provider
+    @role = role
     @roles_mappers = SettingGetter.new(setting_name: 'RoleMapping', provider:).call
   end
 
   def call
-    role = infer_role_from_email(@user_params[:email])
+    email_role = infer_role_from_email(@user_params[:email])
 
     User.new({
       provider: @provider,
-      role:
+      role: email_role || @role
     }.merge(@user_params))
```

Comment:
I think we may want to overwrite the `user_params` with the provider and role instead of the opposite that we're doing,
 There's a room for error from the service perspective where the user_params can overwrite the role and provider,